ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/sse/ClientSSESession.kt

```diff
@@ -19,9 +20,58 @@ public interface SSESession : CoroutineScope {
     public val incoming: Flow<ServerSentEvent>
 }
 
+/**
+ * A Server-sent events session.
+ */
+public interface SSESessionWithDeserialization : CoroutineScope {
+    /**
+     * An incoming server-sent events flow.
+     */
+    public val incoming: Flow<ServerSentEventParsed<String>>
+
+    /**
+     * Deserializer for transforming field `data` of `ServerSentEvent` into desired data object.
+     */
+    public val deserializer: (TypeInfo, String) -> Any
```

Comment:
could we avoid functional fields?

Reply:
@e5l It was an interface, but after https://github.com/ktorio/ktor/pull/4363#discussion_r1793083238 I reverted to the functional field